Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+              one term?”
+            - “If I do this, it crashes. Do you get that as well?”
+            - Activates the Socratic method: great for teaching because
+              it makes people think along with you, not just get
+              battered with assertions
+        - You/we
+              “Why are you showing the number when it's tiny [you
+              idiot]?” vs. “Why are we showing the number when it's
+              tiny? The user can count 3 things, so it's not really
+              necessary.”
+            - Or "this". "This code does X." It's always safe to keep
+              it about the code.
+        - Give compliments! Always say one good thing about the code.
+            - I like to write something encouraging as a top-level
+              comment about the patch in general, while I'm still
+              excited about the actual feature it delivers, before I
+              dig in and discover the code is terrible. "Aw, man, I
+              can't wait to land this. I'm going to love being able to
+              close Flash tabs without reaching for the mouse. [Now
+              let's get the problems fixed so we can merge it.]"
+            - "Good catch!"
+                  These comments would make a good visual, as actual
+                  line comments on a PR.
+                - Also makes a good reply back to the reviewer
+            - "Thank you so much for refactoring this scary mess."
+        - Emoticons
+        - GIFs
+    - TLDR;LGTM
+          A 1000-line review gets 2 comments. A 10-line one gets 5.
+        - Feel free to ask for a more reviewable commit structure,
+          better commit messages, more comments, a couple of paragraphs
+          of prose explaining the approach. If you can't understand it
+          now (with somebody around to answer questions), just imagine
+          the trouble in a year.
+            - When you submit a patch, structure it so you'd be happy
+              to review it.
+            - And definitely read your diffs first before submitting.
+                - Hugely worth it to get a diff viewer you like
+        - Consider not burning yourself out on it; some studies have
+          found a decrease in review effectiveness after about an hour.
+    - Fix the rest of this file while you're at it.
+        - Don't expect them to fix the whole module if they touch part
+          of it.
+            - The idea is for the codebase to be GettingBetter, not
+              BeingPerfect. You can always come back and fix other
+              parts later, and it's okay to point that out, but make
+              clear that it doesn't block the merge (and shouldn't
+              distract them).
+    - Nitpicks
+        - Having a style guide or a linter smooths things along, saves
+          back-and-forth.
+            - But overzealous linters just make me angry. Code is for
+              humans.
```

Comment:
Shouldn't the linting rules be adjusted to just be "zealous" if that's the case? :)

I think linters are really good for two things, and running before asking for review helps with both:
* wastes less time (don't want reviewers spending time pointing out "nits", and reviewees spending time on this)
* keeps codebase consistent

The second point tends to be contentious. It's totally worth discussing whether the linting rules are wrong, or there are problems with the current style, but these should be done outside of the code review and somewhere else ideally (IRC, mailing list, etc)